includes/sanitizers/class-amp-core-theme-sanitizer.php

```diff
@@ -311,6 +319,20 @@ public static function get_supported_themes() {
 		return self::$supported_themes;
 	}
 
+	/**
+	 * Determine if the current request is from a mobile device or not.
+	 *
+	 * @since 2.2
+	 *
+	 * @see MobileRedirection::is_mobile_request()
+	 *
+	 * @return bool True if request is from mobile device Otherwise False.
+	 */
+	public static function is_mobile_request() {
```

Comment:
Is there a way to implement this without resorting to checking the user agent? I worry that checking for whether it is a mobile user agent will not work reliably due to caching layers. This is the same reason why we do client-side redirection by default with JS instead of server-side redirection in PHP.

One possibility would be to use the approach taken for Twenty Seventeen for the sticky menu, where the sanitizer makes a copy of the menu which is shown when it is made sticky. But that may be overkill. It may be worthwhile for each submenu however: include the original `li > ul.sub-menu` but only allow it to be displayed on desktop. Then also include the `li > amp-sidebar` but only show it on mobile. This may involve having two buttons, one that is desktop-specific and one that is mobile-specific.

Reply:
I have updated the implementation and removed the usage of `is_mobile_request()` and used the media query to display AMP navigation for mobile.

However, when bento is enabled ( with the filter `add_filter( 'amp_bento_enabled', '__return_true' );` ) AMP navigation is broken. Currenttly fixing AMP navigation with bento enabled.

Reply:
Broken in what way?

Reply:
| Bento disabled                                                                                                                           | Bento enabled                                                                                                                            |
|------------------------------------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------------|
| <img width="376" alt="image" src="https://user-images.githubusercontent.com/8168027/144975691-f6aaadee-3402-4610-bcc8-50ebdbba7e6a.png"> | <img width="376" alt="image" src="https://user-images.githubusercontent.com/8168027/144975745-2ced64e1-c8d7-4c51-acd0-0423903ed76d.png"> |

I think it is breaking because in bento mode. we are loading script 1.0 version for `amp-sidebar` component. and that may not support `amp-nested-menu`

Reply:
The issue is caused because the bento component for `amp-nested-menu` is not available.